### PR TITLE
add registration email

### PR DESCRIPTION
**Pull request by jamesp-gaia**
re: task for koalas (Koala-127) to add a registration email for users, added server settings rather than djoser based on current use of vanilla DRF endpoints and public registration discussions for biosys

### CODE REVIEW

File: biosys/apps/main/api/views.py [hunk 1]
```diff
@@ -510,6 +510,10 @@ class WhoamiView(APIView):
 
     def get(self, request, **kwargs):
         data = {}
+        print('whoami call')
+        print(request)
+        print(request.user)
+        print(request.user.is_authenticated)
```
Comment (chungvl) [inline biosys/apps/main/api/views.py]: Are these 4 lines debugs?

**Review by chungvl — APPROVED** [review]

File: biosys/apps/main/api/serializers.py [hunk 2]
```diff
@@ -90,6 +91,24 @@ def create(self, validated_data):
                         project = Project.objects.filter(name=project_name).first()
                         if project is not None:
                             project.custodians.add(user)
+
+            # FIXME: A more elegant solution to this would be to extend the use of djoser user registration so that
+            # emails could be handled through the plugin. However at present user creation is handled through a vanilla
+            # DRF endpoint and client applications in production utilise this for user registration, re: discussion on
+            # public user registration for BioSys, so this is used to fulfill requirements for now
+            if settings.SEND_REGISTRATION_CONF:
```
Comment (serge-gaia) [inline biosys/apps/main/api/serializers.py:99]: I really don't like that very much.

File: biosys/apps/main/api/serializers.py [hunk 3]
```diff
@@ -90,6 +91,24 @@ def create(self, validated_data):
...
+                    send_mail(
+                        subject=settings.REGISTRATION_EMAIL_SUBJECT,
+                        message=settings.REGISTRATION_EMAIL_BODY,
+                        html_message=settings.REGISTRATION_EMAIL_BODY,
+                        from_email=(settings.REGISTRATION_EMAIL_FROM + '@' + settings.EMAIL_HOST),
+                        recipient_list=[user.email],
+                        fail_silently=False,
+                    )
+                except Exception as mail_exception:
+                    print(mail_exception)
```
Comment (serge-gaia) [inline biosys/apps/main/api/serializers.py:110]: logger not print!

**Comment by serge-gaia** [comment]
@jamesp-gaia From now assign me as reviewer for biosys server changes.

**Pull request merged**